indigo/indigo/src/main/scala/indigo/platform/events/GlobalEventStream.scala

```diff
@@ -48,23 +48,52 @@ final class GlobalEventStream(
 
     // Storage
     case StorageEvent.FetchKeyAt(index) =>
-      eventQueue.enqueue(StorageEvent.KeyFoundAt(index, storage.key(index)))
+      storage.key(index) match {
+        case Right(err) => eventQueue.enqueue(err)
+        case Left(str)  => eventQueue.enqueue(StorageEvent.KeyFoundAt(index, str))
```

Comment:
Comment further down about Either's being the other way around. In this case I squinted at this for about five minutes convinced there was an error here. There isn't! But it threw me for a while. :smile: